### PR TITLE
Change constants in BHive conversion script

### PR DESCRIPTION
**Pull request by boomanaiden154**
This patch adjusts the constants in the BHive conversion script from the default values added in with the script to the values used within the original BHive paper. These constants were experimentally determined within the BHive paper to be high enough to not underflow/access addresses in the first page yet low enough to not access any addresses above the virtual address space ceiling. Anecdotally, these constants have also seemed to work better than the original ones. In addition, this patch prefixes the memory value with zeroes so that llvm-exegesis is able to assume the correct bit width.

### CODE REVIEW

**Comment by mtrofin** [comment]
> the values used within the original BHive paper. Anecdotally, these constants have also seemed to work better than the original ones.

which are the original counter values, the ones in the original paper or the ones before this change?

File: gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc [hunk 1]
```diff
@@ -29,8 +29,9 @@
 #include "gematria/llvm/llvm_architecture_support.h"
 #include "gematria/utils/string.h"
 
-constexpr uint64_t kInitialRegVal = 0x10000;
-constexpr uint64_t kInitialMemVal = 0x7FFFFFFF;
+constexpr uint64_t kInitialRegVal = 0x12345600;
```
Comment (mtrofin) [inline gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc:32]: add a comment about where the numbers came from, because they are quite odd otherwise

do we understand why they are better, exactly?

File: gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc [hunk 2]
```diff
@@ -64,6 +65,14 @@ int main(int argc, char* argv[]) {
       gematria::ConvertHexToString(kInitialRegVal);
   std::string initial_mem_val_str =
       gematria::ConvertHexToString(kInitialMemVal);
+  // Prefix the string with zeroes as llvm-exegesis assumes the bit width
```
Comment (mtrofin) [inline gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc:68]: this is unrelated?

**Comment by boomanaiden154** [comment]
> which are the original counter values, the ones in the original paper or the ones before this change?

The ones before this change. We're switching to the values in the BHive paper here.

File: gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc (hunk 2 above)
Comment (boomanaiden154) [reply gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc:68]: Slightly unrelated, but necessary to make this patch work properly. We're assuming the memory value has a width of 64 bits, but only setting the first x bits. Exegesis determines the width of the value based on how many characters are in the string, so we need to explicitly print out the leading zeros.

File: gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc (hunk 1 above)
Comment (boomanaiden154) [reply gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc:36]: Comment added along with an explanation for the second question. Essentially they are high enough to avoid accesses in the first page/underflow, but low enough where most accesses are below the virtual address space ceiling. They're not perfect, but anecdotally they work reasonably well.

**Comment by mtrofin** [comment]
> > which are the original counter values, the ones in the original paper or the ones before this change?
> 
> The ones before this change. We're switching to the values in the BHive paper here.

Ack, if you can update the patch description

**Review by mtrofin — APPROVED** [review]
